Source/Core/InputCommon/ControllerEmu/Setting/NumericSetting.h

```diff
@@ -202,4 +208,76 @@ class SettingValue
   mutable InputReference m_input;
 };
 
+template <typename ValueType>
+class SubscribableSettingValue final : public SettingValue<ValueType>
+{
+public:
+  using Base = SettingValue<ValueType>;
+
+  ValueType GetValue() const override
+  {
+    const ValueType cached_value = this->GetCachedValue();
+    if (this->IsSimpleValue())
+      return cached_value;
+
+    const ValueType updated_value = Base::GetValue();
+    if (updated_value != cached_value)
+      TriggerCallbacks();
+
+    return updated_value;
+  }
+
+  void SetValue(const ValueType value) override
+  {
+    if (!this->IsSimpleValue() || (value != this->GetCachedValue()))
+    {
+      Base::SetValue(value);
+      TriggerCallbacks();
+    }
+  }
+
+  struct CallbackID
+  {
+    size_t id;
+
+    bool operator==(const CallbackID&) const = default;
+  };
+
+  using SettingChangedCallback = std::function<void(ValueType)>;
+
+  CallbackID AddCallback(const SettingChangedCallback& callback)
+  {
+    const CallbackID callback_id{.id = m_next_callback_id};
+    ++m_next_callback_id;
+
+    std::lock_guard lock(m_mutex);
```

Comment:
`m_next_callback_id` is being both read from and written to without holding a lock. This could cause duplicate IDs to be assigned (assuming we really do have callers from multiple threads).